allennlp/data/dataset_readers/reading_comprehension/dqa.py

```diff
@@ -0,0 +1,127 @@
+import json
+import logging
+from typing import Any, Dict, List, Tuple
+
+
+from overrides import overrides
+
+from allennlp.common import Params
+from allennlp.common.file_utils import cached_path
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.instance import Instance
+from allennlp.data.dataset_readers.reading_comprehension import util
+from allennlp.data.token_indexers import SingleIdTokenIndexer, TokenIndexer
+from allennlp.data.tokenizers import Token, Tokenizer, WordTokenizer
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("dqa")
+class DQAReader(DatasetReader):
+    """
+    Reads a JSON-formatted QUAC file and returns a ``Dataset`` where the ``Instances`` have four
+    fields: ``question``, a ``TextField``, ``passage``, another ``TextField``, and ``span_start``
+    and ``span_end``, both ``IndexFields`` into the ``passage`` ``TextField``.  We also add a
+    ``MetadataField`` that stores the instance's ID, the original passage text, gold answer strings,
+    and token offsets into the original passage, accessible as ``metadata['id']``,
+    ``metadata['original_passage']``, ``metadata['answer_text_lists'] and ``metadata['token_offsets']``.  
+
+    Parameters
+    ----------
+    tokenizer : ``Tokenizer``, optional (default=``WordTokenizer()``)
+        We use this ``Tokenizer`` for both the question and the passage.  See :class:`Tokenizer`.
+        Default is ```WordTokenizer()``.
+    token_indexers : ``Dict[str, TokenIndexer]``, optional
+        We similarly use this for both the question and the passage.  See :class:`TokenIndexer`.
+        Default is ``{"tokens": SingleIdTokenIndexer()}``.
+    prev_a : Integer, optional 
```

Comment:
In allennlp we prefer unabbreviated names, even if they are a bit longer. Here, I would use `num_previous_answers`, or something similarly easy to scan. If you could update the PR to use this style, that would be great.